File: practice_code/q3.c
#include <stdio.h>


int getSecondLargest(int arr[], int n) {
    
    int largest = -1, secondLargest = -1;

    
    for (int i = 0; i < n; i++) {
        if (arr[i] > largest)
            largest = arr[i];
    }

    
    for (int i = 0; i < n; i++) {
        
       
        if (arr[i] > secondLargest && arr[i] != largest) {
            secondLargest = arr[i];
        }
    }
    return secondLargest;
}

int main() {
    int arr[] = {12, 35, 1, 10, 34, 1}; 
	int n = sizeof(arr)/sizeof(arr[0]);
  
    printf("%d\n", getSecondLargest(arr, n)); 
    return 0; 
}

File: practice_code/q3.py
def getSecondLargest(arr):
    largest = -1
    secondLargest = -1

    
    for num in arr:
        if num > largest:
            largest = num

    
    for num in arr:
        if num > secondLargest and num != largest:
            secondLargest = num

    return secondLargest


arr = [12, 35, 1, 10, 34, 1]
print(getSecondLargest(arr))
